gui-wm/dwl/dwl-0.5-r1.ebuild

```diff
@@ -0,0 +1,86 @@
+# Copyright 2022-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+inherit savedconfig toolchain-funcs
+
+MY_P="${PN}-v${PV}"
+
+WLROOTS_DEP="
+	>=gui-libs/wlroots-0.17:=[libinput,session,X?]
+	<gui-libs/wlroots-0.18:=
+"
+
+if [[ ${PV} == *9999* ]]; then
+	EGIT_REPO_URI="https://codeberg.org/dwl/dwl.git"
+	inherit git-r3
+
+	# 9999-r0: main (latest stable wlroots release)
+	# 9999-r1: wlroots-next (wlroots-9999)
+	case ${PVR} in
+		9999)
+			EGIT_BRANCH=main
+			;;
+		9999-r1)
+			EGIT_BRANCH=wlroots-next
+			WLROOTS_DEP="~gui-libs/wlroots-9999:=[libinput,session,X?]"
+			;;
+	esac
+else
+	SRC_URI="https://codeberg.org/${PN}/${PN}/releases/download/v${PV}/${MY_P}.tar.gz"
+	S="${WORKDIR}/${MY_P}"
+	KEYWORDS="~amd64 ~arm64 ~ppc64 ~x86"
+fi
+
+DESCRIPTION="dwm for Wayland"
+HOMEPAGE="https://codeberg.org/dwl/dwl"
+
+LICENSE="CC0-1.0 GPL-3+ MIT"
+SLOT="0"
+IUSE="X"
+
+CDEPEND="
```

Comment:
For future reference, I try to avoid single-letter "XDEPEND" because it might be used for a future dep class.